sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/ExpressionTypeCheckingSuite.scala

```diff
@@ -811,4 +811,10 @@ class ExpressionTypeCheckingSuite extends SparkFunSuite with SQLHelper with Quer
         "This should have been converted during analysis."))
     )
   }
+
+  test("check that current time is foldable") {
+    val rnd =
+      Rand(Cast(UnixTimestamp(CurrentDate(), Literal("yyyy-MM-dd HH:mm:ss")), IntegerType))
```

Comment:
int: we can simply do `Month(CurrentDate())` to turn date into int.

Reply:
oh, yes, sure.